test/e2e/test/beat/builder.go

```diff
@@ -106,7 +107,13 @@ func newBuilder(name string, suffix string) Builder {
 type ValidationFunc func(client.Client) error
 
 func (b Builder) WithType(typ beatcommon.Type) Builder {
-	b.Beat.Spec.Type = string(typ)
+	typeStr := string(typ)
+	// for Beats we have to use the specific type as there are different Beats images within the one CRD kind.
+	// capitalize the Beat name to be consistent in spelling with the other CRD kinds.
```

Comment:
But you wrote `// Beats images are not supported here we would need 12 additional inputs to accomplish that`?

Reply:
That is true, but I thought it would still support the parameterisation for all apps on the e2e runner level.

Reply:
Got it, let's open this door :+1: